flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/connectors/hive/TableEnvHiveConnectorTest.java

```diff
@@ -165,6 +184,77 @@ public void testDecimal() throws Exception {
 		}
 	}
 
+	/**
+	 * This test is to check whether data can successfully flow from hive source
+	 * into flink system and then into hive sink.
+	 * We especially want to test data types about date here.
+	 *
+	 * <p>TODO: Add timestamp test after fixing the timestamp support of hive connector.
+	 */
+	@Test
+	public void testHiveSourceSink() throws Exception {
+		hiveShell.execute("CREATE DATABASE full_test_db");
+		hiveShell.execute("CREATE TABLE full_test_db.input_tbl (a INT, b STRING, c DOUBLE, d DATE)");
+		hiveShell.insertInto("full_test_db", "input_tbl")
+			.withAllColumns()
+			.addRow(1, "aaa", 1.1, Date.valueOf("2011-01-11"))
+			.addRow(2, "bbb", 2.2, Date.valueOf("2012-02-12"))
+			.addRow(3, "ccc", 3.3, Date.valueOf("2013-03-13"))
+			.commit();
+		hiveShell.execute("CREATE TABLE full_test_db.output_tbl (a INT, b STRING, c DOUBLE, d DATE)");
+
+		TableSchema schema = TableSchema.builder()
+			.field("a", DataTypes.INT())
+			.field("b", DataTypes.STRING())
+			.field("c", DataTypes.DOUBLE())
+			.field("d", DataTypes.DATE())
+			.build();
+
+		ObjectPath sourcePath = new ObjectPath("full_test_db", "input_tbl");
+		CatalogTable sourceTable1 = (CatalogTable) hiveCatalog.getTable(sourcePath);
+		HiveTableSource hiveTableSource1 = new HiveTableSource(new JobConf(hiveConf), sourcePath, sourceTable1);
```

Comment:
I don't think we still need to manually create source and sink in the tests. Similar to other test cases in this class, we can just use sql.